src/hotspot/cpu/ppc/ppc.ad

```diff
@@ -1743,21 +2009,53 @@ uint MachSpillCopyNode::implementation(C2_MacroAssembler *masm, PhaseRegAlloc *r
     else if (src_lo_rc == rc_vs && dst_lo_rc == rc_stack) {
       VectorSRegister Rsrc = as_VectorSRegister(Matcher::_regEncode[src_lo]);
       int dst_offset = ra_->reg2offset(dst_lo);
-      if (masm) {
-        __ addi(R0, R1_SP, dst_offset);
-        __ stxvd2x(Rsrc, R0);
+      if (PowerArchitecturePPC64 >= 9) {
+        if (is_aligned(dst_offset, 16)) {
```

Comment:
If `Matcher::_new_SP` would be aligned to 16 bytes then the offset here would always be also 16 byte aligned. Can be handled separately from this pr.

Reply:
Changing to `_new_SP = OptoReg::Name(align_up(_in_arg_limit, (int)stack_alignment_in_slots()));` should solve it for PPC64.  But I don't know if such a change would be appreciated for other platforms. Some platforms have larger vectors than `StackAlignmentInBytes` and they seem to work fine with 8 Byte aligned vector stack slots. PPC64 actually also works fine with the current alignment, except that it requires an additional add instruction which is a bit ugly.

Reply:
That's correct. The alignment is done in the virtual register file without wasting real stack slots (it corresponds to [pad0](https://github.com/openjdk/jdk/blob/5c73dfc28cbd6801ac85c6685fb8c77aad3ab0b7/src/hotspot/cpu/x86/x86_64.ad#L1921)).
Looks like aarch64 needs [extra instructions](https://github.com/openjdk/jdk/blob/5c73dfc28cbd6801ac85c6685fb8c77aad3ab0b7/src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp#L3763) too if the SP offset isn't aligned.
There's code that aligns the spill masks for vectors. It makes no sense if the spill area isn't aligned properly. I'm planning to open a bug and pr when I find the time...